src/logics/spark/logic.supply.ts

```diff
@@ -0,0 +1,63 @@
+import { Pool__factory } from './contracts';
+import { Service } from './service';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { supportedChainIds } from './configs';
+
+export type SupplyLogicTokenList = [common.Token, common.Token][];
+
+export type SupplyLogicParams = core.TokenToTokenExactInParams;
+
+export type SupplyLogicFields = core.TokenToTokenExactInFields<{ referralCode?: number }>;
+
+export type SupplyLogicOptions = Pick<core.GlobalOptions, 'account'>;
+
+@core.LogicDefinitionDecorator()
+export class SupplyLogic
+  extends core.Logic
+  implements core.LogicTokenListInterface, core.LogicOracleInterface, core.LogicBuilderInterface
+{
+  static readonly supportedChainIds = supportedChainIds;
+
+  async getTokenList() {
+    const service = new Service(this.chainId, this.provider);
+    const reserveTokens = await service.getReserveTokens();
+
+    const tokenList: SupplyLogicTokenList = [];
+    for (const reserveToken of reserveTokens) {
+      if (reserveToken.asset.isWrapped) {
+        tokenList.push([reserveToken.asset.unwrapped, reserveToken.aToken]);
+      }
+      tokenList.push([reserveToken.asset, reserveToken.aToken]);
+    }
+
+    return tokenList;
+  }
+
+  async quote(params: SupplyLogicParams) {
+    const { input, tokenOut } = params;
+    const output = new common.TokenAmount(tokenOut, input.amount);
+    return { input, output };
```

Comment:
As this is copied from `aave-v3`, prefer to keep the same code format without changing empty lines, it helps to future code refactor such as adding `modules/aavev3` similar to `modules/univ3`.

<img width="896" alt="image" src="https://github.com/dinngo/protocolink-logics/assets/4598616/3388dc89-95c2-4345-ba60-ff4bc42622a5">